keanu-project/src/main/java/io/improbable/keanu/tensor/bool/SimpleBooleanTensor.java

```diff
@@ -51,6 +51,18 @@ public SimpleBooleanTensor(boolean constant, int[] shape) {
         Arrays.fill(this.data, constant);
     }
 
+    @Override
+    public BooleanTensor reshape(int... newShape) {
+        IntegerTensor convertedForReshape = IntegerTensor.create(asFlatIntegerArray(), shape);
```

Comment:
No need to change the data for a reshape. Just need to update the `int[] shape` and `int[] stride`.

Reply:
In `ND4JDoubleTensor` the reshape creates a dup and doesn't affect the original, so I've gone for: 

```
    @Override
    public BooleanTensor reshape(int... newShape) {
        if (TensorShape.getLength(shape) != TensorShape.getLength(newShape)) {
            throw new IllegalArgumentException("Cannot reshape a tensor to a shape of different length");
        }
        return new SimpleBooleanTensor(data, newShape);
    }
```